### PR TITLE
MINOR: KAFKA-3176 follow-up to fix minor issues

### PR DESCRIPTION
**Pull request by vahidhashemian**
Co-authored with @ijuma.

### CODE REVIEW

File: core/src/main/scala/kafka/consumer/BaseConsumer.scala [hunk 1]
```diff
@@ -67,22 +67,32 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
...
+        consumer.subscribe(List(topic))
+      case (None, None, None, Some(whitelist)) =>
+        consumer.subscribe(Pattern.compile(whitelist), new NoOpConsumerRebalanceListener())
+      case _ =>
+        throw new IllegalArgumentException("An invalid combination of arguments is provided. " +
+            "Exactly one of 'topic' or 'whitelist' must be provided. " +
+            "If 'topic' is provided, an optional 'partition' may also be provided. " +
+            "If 'partition' is provided, an optional 'offset' may also be provided, otherwise, consumption starts from the earliest message.")
+    }
+  }
```
Comment (vahidhashemian) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: @ijuma I added an additional `case` for when `topic` and `partition` are provided but offset is missing. I also updated the exception message to be more descriptive as you suggested. Please advise if you see an issue. Thanks.

File: core/src/main/scala/kafka/consumer/BaseConsumer.scala [hunk 2]
```diff
@@ -67,22 +67,32 @@ class NewShinyConsumer(topic: Option[String], partitionId: Option[Int], offset:
...
-    else if (whitelist.isDefined)
-      consumer.subscribe(Pattern.compile(whitelist.get), new NoOpConsumerRebalanceListener())
-    else
-      throw new IllegalArgumentException("Exactly one of topic or whitelist has to be provided.")
+    (topic, partitionId, offset, whitelist) match {
+      case (Some(topic), Some(partitionId), Some(offset), None) =>
+        seek(topic, partitionId, offset)
+      case (Some(topic), Some(partitionId), None, None) =>
+        // default to earliest if no offset is provided
+        seek(topic, partitionId, OffsetRequest.EarliestTime)
```
Comment (ijuma) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: Shouldn't the default be `LatestTime`?

Comment (vahidhashemian) [inline core/src/main/scala/kafka/consumer/BaseConsumer.scala]: Correct, and I'll have to similarly update the default for `offsetOpt` in `ConsoleConsumer`. Will update shortly. Thanks.

**Comment by ijuma** [comment]
Thanks @vahidhashemian. I had a look and I think we could clean-up things a bit more in the following way:

https://github.com/ijuma/kafka/commit/8eb57014ab5cfb24099031f19b66e769e5dada9e

It reduces a bit of duplication and handles the case where the user supplies a number that is too large for a `long` value correctly.

What do you think?

**Comment by vahidhashemian** [comment]
@ijuma Your changes look good to me. Should I borrow them to update this PR, or let you take over the patch? Thanks.

**Comment by ijuma** [comment]
Yes, just cherry-pick the commit or integrate in any way you prefer.

**Comment by vahidhashemian** [comment]
Thanks @ijuma. The PR is updated.

**Comment by ijuma** [comment]
LGTM

**Comment by ijuma** [comment]
Merged to trunk.

**Pull request closed without merging**